includes/wc-user-functions.php

```diff
@@ -197,6 +197,12 @@ function wc_paying_customer( $order_id ) {
 function wc_customer_bought_product( $customer_email, $user_id, $product_id ) {
 	global $wpdb;
 
+	$result = apply_filters( 'woocommerce_customer_bought_product', null, $customer_email, $user_id, $product_id );
```

Comment:
This is a good name for filter the final result.
Since this is a pre filter, we could change to something like: `woocommerce_customer_pre_bought_product`